File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        sol = []

        def backtrack(index):
            if len(nums) == index:
                res.append(sol.copy())
                return
            
            #don't take
            backtrack(index + 1)
            
            #take
            sol.append(nums[index])
            backtrack(index + 1)
            sol.pop()
        
        backtrack(0)
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
            for (int i = 0; i < nums.length; i++) {
                for (int k = 0; k < nums.length; k++) {
                    if (i!=k){
                        if (nums[i] == nums[k]) {
                            return true;
                        }
                    }
                }
        }
        return false;
    }
}